src/docs/arch-connection.dox

```diff
@@ -0,0 +1,56 @@
+/*! @arch_page arch-connection Connection
+
+@section arch_conn_def Definition
+
+A connection is a handle to a WiredTiger database instance. The connection has exclusive access to
+the database through a file lock, hence only one connection can be opened at a time. Internally, a
+connection is represented by \c WT_CONNECTION.
+
+@section arch_conn_lifecycle Life cycle
+
+@subsection arch_conn_init Initialization
+
+A connection is initialized when WT_CONNECTION::wiredtiger_open is called by the user application.
+WT_CONNECTION::wiredtiger_open accepts a list of configuration items (see @ref database_config) that
+can be used to enable different WiredTiger features and tune their behavior. Those features are for
+example related to @ref arch-eviction, @ref arch-logging, @ref arch-checkpoint, @ref arch-cache,
+statistics, etc. All the different available configuration settings are described in the
+documentation for WT_CONNECTION::wiredtiger_open.
+
+WT_CONNECTION::wiredtiger_open also performs different sanity checks depending on the configuration
+item "create". When "create" is specified and a database does not already exist, a new database is
+created as well as specific WiredTiger files such as the turtle file and other metadata files. If a
+database already exists, WiredTiger will try to open it and check for the existence of the different
+required WiredTiger files. If "create" is not specified, WiredTiger expects a database where it is
+executed. In case of a failure during WT_CONNECTION::wiredtiger_open, the connection is not created
```

Comment:
expects a database where it is executed -> expects a database to be in a consistent state.

Reply:
Whether it is in a consistent state or not, WT will call RTS on the database, so I don't think it is true that WT expects the database in a consistent state.
And here I cam just talking about the "create" configuration item, when it is not specified, a database should already exist.

Reply:
Maybe we need to somehow indicate that database might be in a corrupted state and by default we don't "fix" it . Not sure what would be the best choice of words here.

Reply:
I tried to rework this part: see 4b33697a24ce220120d7a98ece5f87f1a7abe035